### PR TITLE
add mlflow metric writer

### PR DESCRIPTION
**Pull request by garymm**
add mlflow metric writer

Change-Id: Ief75dd77c24a63b05eec87b5b1f4822abb206abd

### CODE REVIEW

File: jax_loop_utils/metric_writers/mlflow/metric_writer.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import mlflow
+import mlflow.config
+import mlflow.entities
+import mlflow.tracking.fluent
+from jax_loop_utils.metric_writers.interface import (
+    Array,
+    Scalar,
+)
+from jax_loop_utils.metric_writers.interface import (
+    MetricWriter as MetricWriterInterface,
```
Comment (mickvangelderen) [inline jax_loop_utils/metric_writers/mlflow/metric_writer.py:18]: If we need to alias when importing we should just pick a different name

File: jax_loop_utils/metric_writers/mlflow/metric_writer.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+)
+
+
+class MetricWriter(MetricWriterInterface):
```
Comment (mickvangelderen) [inline jax_loop_utils/metric_writers/mlflow/metric_writer.py:22]: Why not `MlflowWriter` to be consistent with other implementations. Having mlflow in the name also provides better feedback in exceptions/debugger/...

File: jax_loop_utils/metric_writers/mlflow/metric_writer.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    """MLflow implementation of MetricWriter."""
+
+    def __init__(self, experiment_name: str):
+        """Initialize MLflow writer.
+
+        Args:
+            experiment_name: Name of the MLflow experiment. If None, uses active experiment.
+        """
```
Comment (mickvangelderen) [inline jax_loop_utils/metric_writers/mlflow/metric_writer.py:30]: doc and type annotation are inconsistent

File: jax_loop_utils/metric_writers/mlflow/metric_writer.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        self._client = mlflow.MlflowClient()
+        experiment_id = mlflow.set_experiment(experiment_name).experiment_id
+        self._run_id = self._client.create_run(experiment_id=experiment_id).info.run_id
```
Comment (mickvangelderen) [inline jax_loop_utils/metric_writers/mlflow/metric_writer.py:33]: We could consider taking the client as an argument, or a run, and provide a convenience function/static method to create a logger `from_experiment_name(experiment: str) -> Self`. 

Motivation would be to not have to replicate the mlflow client, experiment, and run configuration APIs.

**Comment by garymm** [comment]
Closing in favor of #12

File: jax_loop_utils/metric_writers/mlflow/metric_writer.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    def write_scalars(self, step: int, scalars: Mapping[str, Scalar]):
+        """Write scalar metrics to MLflow."""
+        metrics_list: list[mlflow.entities.Metric] = []
+        timestamp = int(time() * 1000)
+        for k, v in scalars.items():
+            metrics_list.append(mlflow.entities.Metric(k, float(v), timestamp, step))
```
Comment (mickvangelderen) [inline jax_loop_utils/metric_writers/mlflow/metric_writer.py:40]: construct list through list comprehension

**Pull request closed without merging**